util/chunk/column.go

```diff
@@ -246,6 +246,64 @@ const (
 	sizeMyDecimal = int(unsafe.Sizeof(types.MyDecimal{}))
 )
 
+// preAlloc allocates space for a fixed-length-type slice and resets all slots to null.
+func (c *Column) preAlloc(length, typeSize int) {
+	c.nullCount = 0
+	nData := length * typeSize
+	if len(c.data) > nData {
+		c.data = c.data[:nData]
+	} else {
+		c.data = make([]byte, nData)
+	}
+
+	nBitmap := (length + 7) >> 3
+	if len(c.nullBitmap) > nBitmap {
+		c.nullBitmap = c.nullBitmap[:nBitmap]
+		for i := range c.nullBitmap {
+			// resets all slots to null.
+			c.nullBitmap[i] = 255
+		}
+	} else {
+		c.nullBitmap = make([]byte, nBitmap)
+	}
+
+	c.length = length
+	c.nullCount = 0
+}
+
+// SetNull sets the rowIdx to null.
+func (c *Column) SetNull(rowIdx int) {
```

Comment:
This function is not used in this commit, can we add it later?

Reply:
It will be used in the next PR, and I add it here to reduce the lines of changed code in the next PR.
Now I add a unit test for it.